acme/acme/messages.py

```diff
@@ -360,8 +368,8 @@ class ChallengeResource(Resource):
     @property
     def uri(self):  # pylint: disable=missing-docstring,no-self-argument
         # bug? 'method already defined line None'
-        # pylint: disable=function-redefined
-        return self.body.uri  # pylint: disable=no-member
+        # pylint: disable=function-redefined,protected-access
+        return self.body._url or self.body._uri  # pylint: disable=no-member
```

Comment:
Can't we just use `self.body.uri` here because of the property in `ChallengeBody` above?